infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -128,7 +129,9 @@
     private long maxConflictWildcard = Address.NOT_FOUND;
 
     private final long defaultCacheSize = Long.MAX_VALUE;
-    private final Cache<Integer, Long> conflictToGlobalTailCache;
+    private final long defaultPerStreamCacheSize = Long.MAX_VALUE;
+    private long perStreamCacheSize = defaultPerStreamCacheSize;
+    private final Cache<UUID, Cache<Integer, Long>> conflictToGlobalTailCache;
```

Comment:
Is there some spec that guarantees that hashCode() is stable across JVMs. please forward the link if you have.  
Yes I just looked the Guava Hasher does not deal with objects that easily. However it does have
the ObjectFunnel thing to pipe in all the relevant members to hash. 

class TestObject {
        String str = "new";
        final int i = 24;
        @Override
        public int hashCode() {
            return Hashing.md5().hashObject(this, new Funnel<TestObject>() {
                @Override
                public void funnel(TestObject testObject, PrimitiveSink primitiveSink) {
                    primitiveSink.putString(str, Charset.defaultCharset());
                    primitiveSink.putInt(i);
                }
            }).hashCode();
        }
    }
    @Test
    public void testHash() {
        System.out.println("THE HASH CODE IS " + new TestObject().hashCode());
    }


You are right, hashCode can be generated post serialization as well and that could be an approach.
Or hashCode can be generated using introspection to create the set of relevant values in the object and then hashing them.

If we had the ConflictParam/Key as a first class interface, we could have our own hash Code which is larger than an integer and also have an equals method.